### ISSUE TITLE
Extend API coverage of fullcalendar events

### ISSUE THREAD

**Issue by dschreij** [CroudSupport/vue-fullcalendar#42]
If you check fullcalendar's documentation for the functions [eventDrop](https://fullcalendar.io/docs/event_ui/eventDrop/) and [eventResize](https://fullcalendar.io/docs/event_ui/eventResize/), you can see that they pass multiple arguments to the event handling function. From all these parameters, the vue component [currently only passes on the event object.](https://github.com/CroudSupport/vue-fullcalendar/blob/master/components/FullCalendar.vue#L102-L108)

I think it would be an improvement to pass on all parameters to the vue event handler. In particular, the removeFunc parameter is very useful for reverting changes to the calendar (e.g. put an event back at its old place after a drag, or resize back to its original size) if the user wasn't allowed to make the change, or an ajax request that is made in the event callback fails. This change should be possible without breaking backward compatibility as the newly passed arguments are simply discared if they are not caught/assigned in the event callback function.

I am willing to work on this and send you a PR, if you are open to this.

**Comment by dschreij** [comment]
By the way, I see you provide the option to alter the method callbacks by overwriting them in the `config` object that can be passed to the component, so this issue is not a critical one. Still, wouldn't it be better to pass on all arguments of fullcalendar's events to vue event handlers?

**Comment by dschreij** [comment]
Well, that is harder than it seemed, as it is really difficult to get the context of the function right. It is very hard to let *self* and *this* point to the vue-calendar component, and thus to call the $emit function to fire the appropriate events.

**Comment by BrockReece** [comment]
Hi @dschreij

Yeah I am always open to PR's and I agree that it would be useful.

As for your third message, shorthand functions do something magic with the scope behind the scenes, it might be easier to pass through a traditional function with a bind method pointing towards `this.$refs.calendar` but I guess I don't really see why you would want to scope the calendar component anyway?

Cheers
Brock

**Comment by dschreij** [comment]
Hi Brock,

The reason for my current solution using shorthands is that I haven't been able to get access to the vue component and its methods (such as $emit) when passing these functions in the config object. I do need to scope the calendar component to get access to its $emit function, am I right?
I did play around with bind() but wasn't able to set the this reference to the full-calendar component.
I tried:
~~~javascript
config:{
	eventDrop: (function(event, delta, removeFunc) {
		this.$emit('event-drop', event, delta, removeFunc);
	}).bind(this.$refs.calendar),
	eventResize: function (event, delta, removeFunc) {
		this.$emit('event-resize', event, delta, removeFunc);
	}).bind(this.$refs.calendar),
}
~~~
but this results in `this` being `undefined`. This is probably due to generating the config object in the `data()` function instead of `created()` or `mounted()` and as such, the components haven't been properly mounted and referenced yet.
I also played around referencing `this` in a `self` var (like you did), but to no avail.

I think the cleanest way of realising this, is to pass all these arguments through the default built-in functions, so I'll send you a PR shortly.
